cmd/object-api-datatypes.go

```diff
@@ -415,6 +417,20 @@ type ListObjectsV2Info struct {
 	Prefixes []string
 }
 
+func concurrentDecryptETag(ctx context.Context, objects []ObjectInfo) {
```

Comment:
Here, you have to provide the S3 HTTP headers. 
The problem with passing `nil` [here](https://github.com/minio/minio/pull/10261/files#diff-db015fc7a9bde5d89c1c062a5741ba50R425) is that in case of SSE-C `getDecryptedETag` would try to decrypt
the ETag but would fail.

Reply:
yeah but this is only meant for Listing @aead

Reply:
Yes, but `getDecrryptedETag` would consider the object metadata as encrypted and then try to decrypt it.
One option would be `crypto.S3.IsEncrypted` instead.

Reply:
@aead but its already handled here

```
         // As per AWS S3 Spec, ETag for SSE-C encrypted objects need not be MD5Sum of the data.                                                                                              
         // Since server side copy with same source and dest just replaces the ETag, we save                                                                                                  
         // encrypted content MD5Sum as ETag for both SSE-C and SSE-S3, we standardize the ETag                                                                                               
         // encryption across SSE-C and SSE-S3, and only return last 32 bytes for SSE-C                                                                                                       
         if crypto.SSEC.IsEncrypted(objInfo.UserDefined) && !copySource {
                 return objInfo.ETag[len(objInfo.ETag)-32:]
         }
```
It won't try to decrypt an SSEC